gammapy/irf/effective_area_table.py

```diff
@@ -0,0 +1,572 @@
+    The interpolation method is defined by self.interpolation_method
+
+    Parameters
+    ----------
+    energ_lo : `~astropy.units.Quantity`
+        Lower energy bin edges vector
+    energ_hi : `~astropy.units.Quantity`
+        Upper energy bin edges vector
+    offset_lo : `~astropy.coordonates.Angle`
+        Lower offset bin edges vector
+    offset_hi : `~astropy.coordonates.Angle`
+        Upper offset bin edges  vector
+    eff_area : `~astropy.units.Quantity`
+        Effective area vector (true energy)
+    eff_area_reco : `~astropy.units.Quantity`
+        Effective area vector (reconstructed energy)
+
+    Examples
+    --------
+    Get effective area vs. energy for a given offset and energy binning:
+
+     .. plot::
+        :include-source:
+
+        import numpy as np
+        from astropy.coordinates import Angle
+        from astropy.units import Quantity
+        from gammapy.irf import OffsetDependentEffectiveAreaTable
+        from gammapy.datasets import load_aeff2D_fits_table
+
+        aeff2D = OffsetDependentEffectiveAreaTable.from_fits(load_aeff2D_fits_table())
+        offset  = Angle( 0.6 , 'degree')
+        energies = Quantity(np.logspace(0,1,60), 'TeV')
+        eff_area = aeff2D.evaluate(offset,energies)
+
+    """
+
+    def __init__(self, energ_lo, energ_hi, offset_lo, offset_hi, eff_area, eff_area_reco):
+        if not isinstance(energ_lo, Quantity) or not isinstance(energ_hi, Quantity):
+            raise ValueError("Energies must be Quantity objects.")
+        if not isinstance(offset_lo, Angle) or not isinstance(offset_hi, Angle):
+            raise ValueError("Offsets must be Angle objects.")
+        if not isinstance(eff_area, Quantity) or not isinstance(eff_area_reco, Quantity):
+            raise ValueError("Effective areas must be Quantity objects.")
+
+        self.energ_lo = energ_lo.to('TeV')
+        self.energ_hi = energ_hi.to('TeV')
+        self.offset_lo = offset_lo.to('degree')
+        self.offset_hi = offset_hi.to('degree')
+        self.eff_area = eff_area.to('m^2')
+        self.eff_area_reco = eff_area_reco.to('m^2')
+
+        # actually offset_hi = offset_lo
+        self.offset = (offset_hi + offset_lo) / 2
+        self.energy = np.sqrt(energ_lo * energ_hi)
+
+        self._prepare_linear_interpolator()
+        self._prepare_spline_interpolator()
+
+        # set to linear interpolation by default
+        self.interpolation_method = 'linear'
+
+    @staticmethod
+    def from_fits(hdu_list):
+        """Create OffsetDependentEffectiveAreaTable from ``GCTAAeff2D`` format HDU list.
+
+        Parameters
+        ----------
+        hdu_list : `~astropy.io.fits.HDUList`
+            HDU list with ``EFFECTIVE AREA`` extension.
+
+        Returns
+        -------
+        eff_area : `OffsetDependentEffectiveAreaTable`
+            Offset dependent Effective Area object.
+        """
+        e_lo = Quantity(
+            hdu_list['EFFECTIVE AREA'].data['ENERG_LO'][0, :], 'TeV')
```

Comment:
Use temp variable `data = hdu_list['EFFECTIVE AREA'].data` so that these statements fit on one line.